test/beast/core/basic_stream.cpp

```diff
@@ -1347,9 +1385,52 @@ class basic_stream_test
                 resolve_results.end(),
                 comparison_function,
                 net::use_awaitable))>);
+
+        static_assert(std::is_same_v<
+            net::awaitable<net::ip::tcp::endpoint>, decltype(
+            stream.async_connect(
+                resolve_results,
+                comparison_function,
+                net::use_awaitable))>);
+
     }
 #endif
 
+    void
+    testConnectionConditionArgs(
+        basic_stream<net::ip::tcp> &stream,
+        net::ip::tcp::resolver::results_type const &results)
+    {
+        struct some_condition
+        {
+            bool operator()(const error_code& ec, net::ip::tcp::endpoint const& ep) const;
+        };
+
+        struct some_handler
+        {
+            void operator()(error_code const& ec, net::ip::tcp::endpoint const& ep) const;
+            void operator()(error_code const& ec, net::ip::tcp::resolver::results_type::const_iterator const& ep) const;
+        };
+
+        //
+        // confirm that both forms of conditional async_connect use a condition funcion with
+        // signature compatible with bool(const error_code& ec, net::ip::tcp::endpoint const& ep)
```

Comment:
This comment doesn't say anything that isn't already said by the code and thus, is extraneous